tests/test_item.py

```diff
@@ -0,0 +1,28 @@
+import os
+import unittest
+import json
+from tempfile import TemporaryDirectory
+from datetime import datetime
+import dateutil
+
+from pystac import *
+from pystac.utils import is_absolute_href
+from tests.utils import (TestCases, RANDOM_GEOM, RANDOM_BBOX, test_to_from_dict)
+
+
+class ItemTest(unittest.TestCase):
+    def test_to_from_dict(self):
+        m = TestCases.get_path('data-files/itemcollections/sample-item-collection.json')
+        with open(m) as f:
+            item_dict = json.load(f)['features'][0]
+        
+        self.assertTrue(test_to_from_dict(Item, item_dict))
```

Comment:
Nice. I originally said this could be a utility, but didn't think about how simple it would be - sorry! So I think just using the unittest.assertDictEquals would be the best move here:

```python
self.assertDictEqual(Item.from_dict(item_dict).to_dict(), item_dict)
```

and removing the utility method.

Reply:
yeah, although I've encountered one edge case when testing the eo class: datetimes. since there are multiple different valid RFC339 strings for a timestamp, the to_dict method doesn't know the exact format of the string in the dict. Do we want to try to ensure that the the timestamp string that each object writes is the exact same as the one it reads?

Reply:
Good call, that makes it tricky. I think we should allow for the datetime string to represent the same thing, but be different encodings.

Given that, it probably makes sense to write our own recursive dict checking code that handles these and any other exceptional cases that we may come across. I'd probably aim for it not too be too robust (no need to spend a ton of time engineering test code), but something that can handle the datetime case and have a place if we run into other exceptions to the `==` rule.

Reply:
ok, so I'll just expand the `test_to_from_dict` method and leave it in the same place

Reply:
Makes sense.

I will say, with that method just returning a boolean, there's not going to be a lot of help to the person debugging the test (including future you) if it'll fail with `False not equal to True`.

Since this is a test utility, you might want to just pass in `self` and do the asserts at the property level to check equality of the dictionary items, so that it'll fail with the actual values that didn't match.

You can also use the `msg` parameter in [assertEqual](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertEqual) to give the dict key that it failed on.

Reply:
this makes sense

but i'm not sure how to use the msg param to report the key that it failed on. How would I access that within the assert method?

Reply:
After thinking about it, if you just use `assertDictEquals`, you should be able to avoid recursing through each property and not have to use the `msg` param.